spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -113,17 +112,67 @@ public ParameterDescription describe(MethodParameter parameter) {
 
 	@Override
 	public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext context) {
-		return null;
+		// DevNote: in Legacy the user is either completing an argument (-- prefix) or the value of the argument (e.g. no arity to consider).
```

Comment:
Comment wrt arity: not entirely true, as there is the `specifiedDefaultValue` case where you can have a pseudo arity=0 case